src/language/worker.ts

```diff
@@ -92,98 +96,85 @@ const executeAll = async (
   }
 };
 
-const findAndParse = async (
-  serviceProvider: CliServiceProvider,
-  databaseName: string,
-  collectionName: string
-) => {
-  const documents = await serviceProvider
-    .find(databaseName, collectionName, {}, { limit: 1 })
-    .toArray();
-
-  if (documents.length === 0) {
-    return [];
-  }
-
-  try {
-    const schema = await parseSchema(documents);
-
-    if (!schema || !schema.fields) {
-      return [];
-    }
-
-    const fields = schema.fields.map((item) => ({
-      label: item.name,
-      kind: CompletionItemKind.Field,
-    }));
-
-    return fields;
-  } catch (parseError) {
-    return [];
-  }
-};
-
+/**
+ * Get field names for autocomplete.
+ * @public
+ */
 const getFieldsFromSchema = async (
   connectionString: string,
-  connectionOptions: any,
+  connectionOptions: ConnectionOptions,
   databaseName: string,
   collectionName: string
-): Promise<any> => {
-  try {
-    const serviceProvider: CliServiceProvider =
-      await CliServiceProvider.connect(connectionString, connectionOptions);
+): Promise<[unknown, string[]?]> => {
+  let serviceProvider: CliServiceProvider;
+  let result: string[] = [];
 
-    const result = await findAndParse(
-      serviceProvider,
-      databaseName,
-      collectionName
+  try {
+    serviceProvider = await CliServiceProvider.connect(
+      connectionString,
+      connectionOptions
```

Comment:
Does this work? The `ConnectionOptions` type from the driver is not something that an application should normally use, and it’s also not what `CliServiceProvider.connect` expects

Reply:
Replaced with MongoClientOptions defined here in the begging of the file. But answering your question if it worked, weirdly yes)